scripts/build_material.py

```
@@ -0,0 +1,254 @@
+    if parsed is not None and not isinstance(parsed, dict):
+        raise ValueError(
+            "人間メモのフロントマターはマッピングである必要があります: "
+            f"{type(parsed).__name__} を検出しました"
+        )
+
+    front = parsed if isinstance(parsed, dict) else {}
+    return front, body.strip()
+
+
+def derive_created(stem: str) -> str:
+    """ファイル名 stem の先頭 `YYYY-MM-DD` から ISO 日時を導出する．
```

Ruff (0.15.15)

[warning] Docstring contains ambiguous `．` (FULLWIDTH FULL STOP). Did you mean `.` (FULL STOP)?

(RUF002)